server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/models/agents/agents.js

```diff
@@ -44,6 +45,24 @@ define(['mithril', 'lodash', 'string-plus',
       });
     };
 
+    var agentsWithState = _.memoize(function (state) {
+      return self.filterAgent(function (agent) {
+        return agent.agentConfigState() === state;
+      }).length;
+    });
```

Comment:
Instead of declaring the variable self, I think it's better to ```bind(this)```

```
    var agentsWithState = _.memoize((function (state) {
      return this.filterAgent(function (agent) {
        return agent.agentConfigState() === state;
      }).bind(this)).length;
   });
```